privval/ipc.go

```diff
@@ -0,0 +1,218 @@
+package privval
+
+import (
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	cmn "github.com/tendermint/tendermint/libs/common"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/types"
+)
+
+// IPCPV implements PrivValidator, it uses a unix socket to request signatures
+// from an external process.
+type IPCPV struct {
```

Comment:
It makes sense to encode the protocol in with the advent of alternative implementations, but we should have all implemenetations consistent. Means that `SocketPV` should likely become `TCPPV`. Which in turn reads horribly, maybe we can consider `IPCVal` and `TCPVal`.